### PR TITLE
Add queue-worker metrics to metrics page

### PR DESCRIPTION
**Pull request by welteki**
Signed-off-by: Han Verstraete (OpenFaaS Ltd) <han@openfaas.com>

## Description

Add queue-worker metrics to metrics page.

## Motivation and Context

Add a reference for user to see what metrics are available on the queue-worker.

## How Has This Been Tested?

Build and inspect page locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

File: docs/architecture/metrics.md [hunk 1]
```diff
@@ -83,4 +83,13 @@ The [FaaS Provider](/architecture/faas-provider) is the back-end API used by oth
 | `provider_http_request_duration_seconds`     | histogram  | Seconds spent serving HTTP requests | `method`, `path`, `code`   | Community Edition  |
 | `provider_http_requests_total`               | counter    | The total number of HTTP requests   | `method`, `path`, `code`   | Community Edition  |
 
-The `http_request*` metrics record the latency and statistics of `/system/*` routes. Part of this information is also recorded in the metrics for the Gateway component. The purpose of exposing separate metrics on the provider component is to show the count of calls, to show efficiency, and to show the duration for performance testing, along with errors to flag unseen issues.
\ No newline at end of file
+The `http_request*` metrics record the latency and statistics of `/system/*` routes. Part of this information is also recorded in the metrics for the Gateway component. The purpose of exposing separate metrics on the provider component is to show the count of calls, to show efficiency, and to show the duration for performance testing, along with errors to flag unseen issues.
+
+## JetStream queue-worker
```
Comment (alexellis) [inline docs/architecture/metrics.md:88]: I've started calling this "JetStream for OpenFaaS", can we use that heading?

Then underneath explain they are emitted from the queue-worker.

File: docs/architecture/metrics.md [hunk 2]
```diff
@@ -83,4 +83,13 @@ The [FaaS Provider](/architecture/faas-provider) is the back-end API used by oth
...
+The queue-worker for NATS JetStream exposes metrics to help you get insight in the behavior of your OpenFaaS queues.
+
+| Metric                                  | Type       | Description                                     | Labels       | Edition     |
+| ----------------------------------------| ---------- | ------------------------------------------------| -------------|-------------|
+| `queue_worker_pending_messages`         | gauge      | Messages waiting to be completed                | `queue_name` | Pro Edition |
```
Comment (alexellis) [inline docs/architecture/metrics.md:93]: Amount of messages waiting to be processed on given `queue_name`

File: docs/architecture/metrics.md [hunk 3]
```diff
@@ -83,4 +83,13 @@ The [FaaS Provider](/architecture/faas-provider) is the back-end API used by oth
...
+| `queue_worker_messages_processed_total` | gauge      | Total number of messages processed              | `queue_name` | Pro Edition |
```
[the remark this answers is not in the document]
Comment (welteki) [reply docs/architecture/metrics.md:94]: Next to the `queue_name` label it has `app`, `instance`, `job`, `kubernetes_namespace`, `kubernetes_pod_name` and `pod_template_hash` labels.

File: docs/architecture/metrics.md [hunk 4]
```diff
@@ -83,4 +83,13 @@ The [FaaS Provider](/architecture/faas-provider) is the back-end API used by oth
...
+| `queue_worker_messages_submitted_total` | gauge      | Total number of messages submitted to the queue | `queue_name` | Pro Edition |
```
[the remark this answers is not in the document]
Comment (welteki) [reply docs/architecture/metrics.md:95]: Same a here: https://github.com/openfaas/docs/pull/301#discussion_r927722606

File: docs/architecture/metrics.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (welteki) [reply docs/architecture/metrics.md:94]: Should the `kubernetes_pod_name` be added to the table as well?

[the remark this answers is not in the document]
Comment (alexellis) [reply docs/architecture/metrics.md]: I think it may help to show it here.

[the remark this answers is not in the document]
Comment (alexellis) [reply docs/architecture/metrics.md]: @welteki have you resolved this? If so, please click "resolve conversation"

**Pull request merged**